app/common/renderer/actions/SessionInspector.js

```
@@ -328,12 +330,51 @@ export function applyClientMethod(params) {
       return commandRes;
     } catch (error) {
       log.error(error);
+      if (getState().inspector.autoSessionRestart) {
+        const restartSes = restartSession(error, params);
+        return await restartSes(dispatch, getState);
+      }
       showError(error, {methodName: params.methodName, secs: 10});
       dispatch({type: METHOD_CALL_DONE});
     }
   };
 }
 
+export function restartSession(error, params) {
+  return async (dispatch, getState) => {
+    const isApplyError = error.message === "Cannot read properties of undefined (reading 'apply')";
```

Could you explain the situation in which you get this error? I was attempting to replicate various interactions after disconnecting the device under test, but only got the session crashed/socket hang up error.

---

Thanks, I can reproduce this now. Unfortunately this reveals that proper handling of device disconnection would require additional complexity: in your first example, the Appium server log returned HTTP 500, whereas in the second it was HTTP 404. Since the second case does not involve device disconnection, relying on the error you see in the GUI would result in this case incorrectly triggering a session reload.

My guess is that since the HTTP 500 error is coming from the `elementFind` command, it would likely need to be handled in [`inspector-driver.js:206`](https://github.com/appium/appium-inspector/blob/main/app/common/renderer/lib/appium/inspector-driver.js#L206), by checking the error code. If it's 500, you can propagate the error further, which should then be caught by `applyClientMethod` and `restartSession`. Implementing it this way would also trigger the session reload as soon as the element is selected, without needing to click Tap.